resources/rules/post_identifier_merge/temp/abstraction/gene_or_gene_product_abstraction.clj

```diff
@@ -0,0 +1,20 @@
+;; ----------------------------------------------------
+;; --------- Gene or Gene Product Abstraction ---------
+;; ----------------------------------------------------
+;;TODO the restrictions need to be properly reified
+
+`{:name "gene-or-gene-product-abstraction"
+  :description "This rule creates an absraction for referring to genes or gene products"
+  ;; this needs to be hashed and have proper classes
+  :head ((?/gene_product rdfs/subClassOf ?/gene_or_gene_product))
+  
+  :body ((?/gene rdf/type kbio/GeneSpecificGClass)
+         (?/gene rdfs/subClassOf ?/gene_or_gene_product)
+         (?/gene_or_gene_product rdf/type kbio/GeneSpecificGorGPClass)
+
+         (?/has_indriect_template_restriction owl/someValuesFrom ?/gene)
+         (?/has_indriect_template_restriction owl/onProperty kso/has_indirect_template)
```

Comment:
For any relation with a non-ccp namespace, we should create a new relation within the ccp-extension ontology, or find a suitable replacement. Our choices for kso/has_indirect_template are to create a relation with identifier SO_EXT_0000001 in the ccp-extension ontology, or to use the protein ontology relation: obo/pr#has_gene_template. I'm inclined to use the latter since it already exists. We just need to make sure we use the same one for the rule that asserts the has_indirect_template relations in the rule head.